src/commons/api/ca.rs

```diff
@@ -525,11 +578,23 @@ impl ObjectName {
     }
 
     pub fn aspa(customer: Asn) -> Self {
-        ObjectName(format!("{}.asa", customer))
+        ObjectName(format!("{}.asa", customer).into())
     }
 
     pub fn bgpsec(asn: Asn, key: KeyIdentifier) -> Self {
-        ObjectName(format!("ROUTER-{:08X}-{}.cer", asn.into_u32(), key))
+        ObjectName(format!("ROUTER-{:08X}-{}.cer", asn.into_u32(), key).into())
+    }
+}
+
+impl From<&uri::Rsync> for ObjectName {
```

Comment:
hm okay.. so this actually only use in `CertInfo`. I was deriving it from the certificate's uri because this saves some (tiny amount of) memory. Otoh that incurs a runtime cost when we publish, and it would a bit late discover that the names cannot be derived from the URIs.

I have now ported the code into `CertInfo::create`:
```
let name = {
     let path = uri.path();
     let after_last_slash = path.rfind('/').unwrap_or(0) + 1;
     // certificate file names must end with .cer and have at least
     // one more character before the .cer filename extension - i.e. we
     // expect 5 characters after the last slash.
     if !path.ends_with(".cer") || path.len() < after_last_slash + 5 {
           Err(InvalidCert::Uri(uri.clone()))
     } else {
           Ok(ObjectName(path[after_last_slash..].into()))
      }
}?;
```

This is not expected to ever fail, but.. if there is an issue then this would be the moment to fail..

There is no unit test for this, but the behaviour is tested extensively by the functional tests that check for the presence of named files.

Reply:
👍